### PR TITLE
fix: remove node buffers

### PR DESCRIPTION
**Pull request by achingbrain**
Updates to the latest changes from interface-datastore and datastore-core

Depends on:

- [x] https://github.com/ipfs/interface-datastore/pull/43
- [x] https://github.com/ipfs/js-datastore-core/pull/27

BREAKING CHANGE: only uses Uint8Arrays internally

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -216,22 +217,31 @@ class FsDatastore extends Adapter {
...
+      for await (const file of files) {
+        try {
+          const buf = await fsReadFile(file)
+
+          yield {
+            key: this._decode(file),
+            value: buf
+          }
+        } catch (err) {
+          console.info(err)
```
Comment (Gozala) [inline src/index.js:242]: 🚨 It looks like this would start swallowing errors, which seems unrelated to `Uint8Array` change.
📝 Should this be `console.error(error)` (or `console.warn`) instead ?

Comment (achingbrain) [reply src/index.js:242]: Oops, no, this shouldn't be in there at all.  It's related to a new test added to interface-datastore that mutates the datastore while running a query (a block is deleted).  To pass the test we don't have to take the mutation into account but we shouldn't explode either.
